cmd/eib/main.go

```diff
@@ -24,43 +25,176 @@ import (
 )
 
 const (
-	argConfigFile = "config-file"
-	argConfigDir  = "config-dir"
-	argBuildDir   = "build-dir"
-	argValidate   = "validate"
+	argDefinitionFile = "config-file"
+	argConfigDir      = "config-dir"
+	argBuildDir       = "build-dir"
+	argValidate       = "validate"
 )
 
-func processArgs() (*image.Context, error) {
-	var (
-		configFile   string
-		configDir    string
-		rootBuildDir string
-		validate     bool
-	)
-
-	flag.StringVar(&configFile, argConfigFile, "", "name of the image configuration file")
-	flag.StringVar(&configDir, argConfigDir, "", "full path to the image configuration directory")
-	flag.StringVar(&rootBuildDir, argBuildDir, "", "full path to the directory to store build artifacts")
-	flag.BoolVar(&validate, argValidate, false, "if specified, the image definition will be validated but not built")
+const (
+	exitCodeDefinitionParse   = 10
+	exitCodeDefinitionInvalid = 11
+	exitCodeConfigDir         = 12
+	exitCodeBuildDir          = 13
+	exitCodeRpmDependencies   = 14
+)
+
+type CLIArguments struct {
+	definitionFile string
+	configDir      string
+	rootBuildDir   string
+	validate       bool
+}
+
+func main() {
+	cliArguments := parseCliArguments()
+
+	buildDir, combustionDir, err := build.SetupBuildDirectory(cliArguments.rootBuildDir)
+	if err != nil {
+		audit.Auditf("The build directory could not be setup under the configuration directory '%s'.", cliArguments.configDir)
+		audit.Audit("Reason:")
+		audit.AuditError(err)
```

Comment:
The main issue I find with this sort of logging is that if (or rather when) the writer behind these changes, then it'd be a hassle to fix the messages e.g. if instead of just writing plain text to stdout via `fmt.Println` we switch to `log.Println` which includes timestamps, the "Reason" and the error logs will be separate log messages with different timestamps.

As a bonus point string formatting which is not currently detectable because it's abstracted under a method call, will instead be in place and very easy to maintain e.g. it's very easy to call `audit.AuditfAndLog` without specifying a `%s` (or any other) formatter.

I'd suggest to use a `strings.Builder` similarly to its usage for the validation and build and a log a single message instead of multiple ones.